dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/NaiveDevirtualisation.java

```diff
@@ -97,16 +92,16 @@ private void findAndTransformAddressTakenFunctions(
         }
     }
 
-    private boolean assignAddressToFunction(Function func, Map<Function, IntLiteral> func2AddressMap) {
-        final IntegerType ptrType = TypeFactory.getInstance().getArchType();
-        final ExpressionFactory expressions = ExpressionFactory.getInstance();
-        if (!func2AddressMap.containsKey(func)) {
-            logger.debug("Assigned address \"{}\" to function \"{}\"", nextAvailableFuncAddress, func);
-            func2AddressMap.put(func, expressions.makeValue(nextAvailableFuncAddress, ptrType));
-            nextAvailableFuncAddress += 8;
-            return true;
+    private boolean assignAddressToFunction(Function func, Map<Function, MemoryObject> func2AddressMap) {
+        if (func2AddressMap.containsKey(func)) {
+            return false;
         }
-        return false;
+
+        logger.debug("Assigned address to function \"{}\"", func);
```

Comment:
nitpicking: this should probably come last, in case some of the following operations failed (which should never be the case, but still)